docs/docsite/rst/presentations/ansible-testing.rst

```diff
@@ -0,0 +1,438 @@
+Ansible Testing
+===============
+
+
+Current as of Ansible 2.3 (May 2017)
+
+Who am I?
+=========
+
+John Barker
+
+* Principal Engineer, Ansible by Red Hat
+* Hats: Ansible Networking, QA and community
+* Freenode/GitHub: `gundalow <https://github.com/gundalow>`_
+
+.. container:: handout
+
+   * Based in the UK
+   * ~ 1 year
+
+Ansible Testing
+===============
+
+* History of testing Ansible
+* How `you` can develop tests
+* Where to find out more
+
+.. container:: handout
+
+   * Set the scene
+
+     * Quick taster, give you enough to know what's possible and point you in the right direction
+     * Important: Reduce regression, help support Python 3
+
+   * Tailor content:
+
+     * Q: Show of hands for version, devel, 2.3, -> 1.9? - upgrade!
+     * Q: Who's raised a PR, should be getting easier on each release, especially modules
+
+   * ReStructured Text: Slides will be made available online, which include speakers notes
+   * All docs & module docs can be edited in GitHub UI, no knowledge of Git needed!
+   * Questions welcome at any point - Will ask Mark to make a note
+
+
+Requirements
+============
+
+What do we want from testing Ansible:
+
+* Prevent regressions & speed up development
+* Increase code quality
+* Better readable code, improve collaboration
+
+* Ability to run same tests locally as run by CI (Shippable)
+* Different OS, Python versions (2.4+, 3.5+)
+* Reasonable run time
+
+.. container:: handout
+
+   * Local run - common request from the community
+   * Increasing the matrix of Python & OS versions vs timely feedback
+   * Main aim: Stable product, find issues sooner, speed up PR work flow
+
+Solution: ansible-test
+======================
+
+* Shippable
+* Different OS & Python: Docker & tox
+* Local runs: Shippable is just a wrapper
+* Duration: Inspect git diff. File to test mapping.
+* Nightly runs with code coverage analysis.
```

Comment:
Remove periods at the end of phrases to stay consistent with other two bullets:
* Duration: Inspect git diff file-to-test mapping
* Nightly runs with code coverage analysis